### ISSUE TITLE
terminology – roads, streets, edges, lines, links, etc.

### ISSUE THREAD

**Issue by jGaboardi** [uscuni/neatnet#141]
Within the docs, comments, and code itself we refer variously to roads, streets, edges, lines, links, etc.

In our last meeting during the discussion where we decided the [final package name](https://github.com/uscuni/neatnet/issues/64) we mentioned that it is nice to be agnostic in terms of types of network (streets, bike paths, waterways, etc.).

So considering this, should we be more deliberate in terminology. Perhaps (trying to) stick with the more abstract [edges, lines, links] over the more specific [roads, streets] where possible?

**Comment by anastassiavybornova** [comment]
@jGaboardi  good point, suggestion from @martinfleis  (and I agree) to first fix terminology everywhere in the paper and once that is done, propagate here

**Comment by jGaboardi** [comment]
Good idea.

**Comment by jGaboardi** [comment]
@martinfleis @anastassiavybornova 

Do either of you have any thoughts/suggestions on this now that the code base is more mature and the pre-print is in the wild?

**Comment by martinfleis** [comment]
Good question. Do we actually use this constantly in the paper?

**Comment by jGaboardi** [comment]
> Do we actually use this constantly in the paper?

Also a good question...

**Comment by anastassiavybornova** [comment]
it seems that for now, _in the paper_, some streamlining might be necessary (lmk if you think this should be done before arxiv v2?):
* we use "edges", not "links", to refer to the non-nody network parts 
* mostly "streets", but sometimes "roads" (interchangeably, as far as i can judge)
* we use "segments" in compound terms a lot ("road segment", "street segment", but also ""network edge segment") 


-- and i think it makes sense to _in the package_ be more abstract as @jGaboardi  suggests above, maybe we could:
* use "edge" whereever the context is the network object (graph) 
* use "line" whereever the context is the geometrical object (linestring, etc.)

**Comment by martinfleis** [comment]
Okay, I think we need to clean both, before arxiv v2.

I don't have a clear suggestion how as of now, but it is late here.

**Comment by jGaboardi** [comment]
James' quick opinions based on @anastassiavybornova's ideas above:

* since "street" is in the title we stick to that where possible, both in the paper and code base
* favor "edge" over "link" 
* try to stick to using the compound only in terms of the geometric primitive - "line segment" - when talking about a portion of a linestring, not the full linestring


Besides the manuscript and docstring within the code base we should also review:
* the actual variable names & comments in the code
* the README
* the notebooks in `docs/source/*`

**Comment by martinfleis** [comment]
I tried to clean this in the manuscript. Please check. I tried to ensure that we use "street" and not "road", that we use "edge" when referring to a single piece from node to node and I kept "segment" only when defining what an edge is from the perspective of a street.

**Comment by jGaboardi** [comment]
As I start with the codebase part of this, do we want to provide a deprecation period for any arguments, e.g.  [`roads`](https://github.com/uscuni/neatnet/blob/84a1dcac84b50d88be1eed294d3e00546a6c5f73/neatnet/simplify.py#L660) in `neatify()`?

As far as I can tell, it's only positional argument names that need adjusting in public-facing functions and we have the `*` signatures in those, so maybe OK to simply do this at once without a deprecation?

**Comment by martinfleis** [comment]
Just flip the switch.

**Comment by jGaboardi** [comment]
> Just flip the switch.

Alright then, it's done in #234
